products/bash/setup-demos.sh

```diff
@@ -155,16 +161,45 @@ function merge_addon {
   echo ${ADDON_ARRAY_JSON}
 }
 
+function update_conditions() {
+  MESSAGE=${1}
+  REASON=${2}            # command type
+  CONDITION_TYPE="Error" # for the type in conditions
+  TIMESTAMP=$(date -u +%FT%T.%Z)
+
+  $DEBUG && echo -e "$info update_conditions(): message($MESSAGE) - reason($REASON) - conditionType($CONDITION_TYPE) - timestamp($TIMESTAMP)"
+
+  # update condition array
+  CONDITION_TO_ADD=$(echo $CONDITION_ELEMENT_OBJECT | jq -r '.message="'"$MESSAGE"'" | .status="True" | .type="'$CONDITION_TYPE'" | .lastTransitionTime="'$TIMESTAMP'" | .reason="'$REASON'" ')
+  # add condition to condition array
+  STATUS=$(echo $STATUS | jq -c '.conditions += ['"${CONDITION_TO_ADD}"']')
+  $DEBUG && echo -e "\n$info [INFO] Printing the conditions array" && echo $STATUS | jq -r '.conditions'
+}
+
+function update_phase() {
+  PHASE=${1} # Pending, Running or Failed
+
+  $DEBUG && divider && echo -e "$info [INFO] update_phase(): phase($PHASE)"
+
+  STATUS=$(echo $STATUS | jq -c '.phase="'$PHASE'"')
+
+  # if the phase is failed, then exit status (case insensitive checking)
+  if echo $PHASE | grep -iqF failed; then
+    divider
+    exit 1
+  fi
```

Comment:
Maybe this shouldn't be done in `update_phase`. Let the install continue and attempt to install all addons, then do a check after the loop and exit if phase is failed?

Reply:
Done